OSX nightly build C compiler issue

CMOR's OSX nightly build is not being built due to an error with the C compiler.
```
checking whether the C compiler works... no
configure: error: in `/Users/distiller/project/conda_build/cmor_1562609657582/work':
configure: error: C compiler cannot create executables
See `config.log' for more details
Traceback (most recent call last):
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 445, in main
    execute(sys.argv[1:])
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 436, in execute
    verify=args.verify, variants=args.variants)
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 2313, in build_tree
    notest=notest,
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1461, in build
    cwd=src_dir, stats=build_stats)
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/utils.py", line 397, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/Users/distiller/project/workspace/test_macos_cmor/miniconda/lib/python3.7/site-packages/conda_build/utils.py", line 377, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/Users/distiller/project/conda_build/cmor_1562609657582/work/conda_build.sh']' returned non-zero exit status 77.
Exited with code 1
```
@doutriaux1 @dnadeau4 Have you encountered this error?  Could you provide a solution?

--- a/recipes/cmor/build.sh
+++ b/recipes/cmor/build.sh
@@ -13,6 +13,7 @@ fi
 if [[ `uname` == "Linux" ]]; then
     export LDSHARED_FLAGS="-shared -pthread"
 else
+    # For OSX builds, reset the include and linker root path for clang
     export CFLAGS="-Wl,-syslibroot / -isysroot / ${CFLAGS}"
     export LDSHARED_FLAGS="-bundle -undefined dynamic_lookup"
 fi

--- a/recipes/cmor/build.sh
+++ b/recipes/cmor/build.sh
@@ -1,4 +1,4 @@
-export CFLAGS="-Wall -g -m64 -pipe -O2  -fPIC"
+export CFLAGS="-Wall -g -m64 -pipe -O2 -fPIC ${CFLAGS}"
 export CXXLAGS="${CFLAGS}"
 export CPPFLAGS="-I${PREFIX}/include"
 export LDFLAGS="-L${PREFIX}/lib"
@@ -13,6 +13,7 @@ fi
 if [[ `uname` == "Linux" ]]; then
     export LDSHARED_FLAGS="-shared -pthread"
 else
+    export CFLAGS="-Wl,-syslibroot / -isysroot / ${CFLAGS}"
     export LDSHARED_FLAGS="-bundle -undefined dynamic_lookup"
 fi
 

--- a/scripts/conda_upload.sh
+++ b/scripts/conda_upload.sh
@@ -33,7 +33,7 @@ fi
 conda config --set anaconda_upload no
 mkdir ~/conda-bld
 if [ `uname` == "Linux" ]; then
-    conda install -n root -q anaconda-client "conda-build<3.3"
+    conda install -n root -q anaconda-client conda-build
 fi
 if [ `uname` == "Darwin" ]; then
     # fix conda and anaconda-client conflict

--- a/setup.py.in
+++ b/setup.py.in
@@ -47,6 +47,9 @@ for p in include_dirs:
       ld.append(p)
 include_dirs=ld
 
+if sys.platform == "darwin":
+  os.environ["CFLAGS"]="-Wl,-syslibroot / -isysroot /"
+
 print('Setting up python module with:')
 print('libraries:',libraries)
 print('libdir:',library_dirs)

--- a/setup.py.in
+++ b/setup.py.in
@@ -47,6 +47,7 @@ for p in include_dirs:
       ld.append(p)
 include_dirs=ld
 
+# For OSX builds, reset the include and linker root path for clang
 if sys.platform == "darwin":
   os.environ["CFLAGS"]="-Wl,-syslibroot / -isysroot /"
 

